internal/p2p/router.go

```diff
@@ -417,7 +417,7 @@ func (r *Router) routeChannel(
 
 func (r *Router) numConcurrentDials() int {
 	if r.options.NumConcurrentDials == nil {
-		return runtime.NumCPU()
+		return runtime.NumCPU() * 32
```

Comment:
Does this need to be a factor of `NumCPU()`. Can we simplify this to a constant.

Reply:
We can, although, since the goal of the pool is to limit the amount of resource consumption, it makes sense to scale it by `NumCPU`. `NumCPU` describes the number of resources we have available. On larger machines we can start more routines and still consume a similar percent of available resources. On smaller machines, we want to start fewer to ensure we don't starve out other processes.

Reply:
I understood this, but this has a direct effect on dialling speed. I don't think this operation is going to be dominant in overall CPU bandwidth (I could be wrong) and it seems odd that larger machines dial faster than smaller machines (can't they all dial at a similar rate)

Reply:
Having an upper bound makes sense, but there should not be much difference in overall CPU consumption—on this workload it will not pin even a single CPU to have multiple goroutines in flight on this task.